fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -0,0 +1,195 @@
+
+                if not module_files:
+                    raise ValueError("No gRPC protocol buffer files found in your interface_dir for the service '{0}'. Please copy the {0}_pb2.py and {0}_pb2_grpc.py files to your interface_dir".format(_grpc_package_name))
+                else:
+                    for file_name in module_files:
+                        if not (file_name.endswith('.py') or file_name.endswith('.pyc')) and 'pb' not in file_name:
+                            raise ValueError("Invalid file '{0}' found in your interface_dir for the service '{1}'. Please only copy the {1}_pb2.py and {1}_pb2_grpc.py files to your interface_dir".format(file_name, _grpc_package_name))
+
+                grpc_interface_module_list = grpc_helper_obj.get_grpc_interface_module(module_files,
+                                                                                       _grpc_interface_file_dir,
+                                                                                       _grpc_package_name)
+                log.debug(grpc_interface_module_list)
+
+                grpc_stub_tuple = grpc_helper_obj.get_grpc_class(grpc_interface_module_list, 'stub')
+                if not grpc_stub_tuple:
+                    raise ValueError("No gRPC 'stub' class found in the protocol buffer files in the interface_dir")
+
+                grpc_request_tuple = grpc_helper_obj.get_grpc_class(grpc_interface_module_list,
+                                                                    _grpc_request_method_name.lower())
+            except Exception as error_msg:
+                raise ValueError("Failed to load the gRPC protocol buffer files:\nERROR: {0}".format(error_msg))
+
+            # Converting resilient function input data into json object
+            try:
+                grpc_function_json_data = json.loads(grpc_function_data)
+            except Exception:
+                raise FunctionError(u"Function input 'grpc_function_data' must be a JSON formatted String: {0}".format(grpc_function_json_data))
+
+            # Checking Communication type
+            if _grpc_communication_type.lower().strip() == "unary":
+                try:
+                    # Connecting to gRPC Server with given Authentication type
+                    stub_class_obj, grpc_channel_obj = grpc_helper_obj.gRPC_Connect_Server(_grpc_secure_connection,
+                                                                                           grpc_channel,
+                                                                                           grpc_stub_tuple[1],
+                                                                                           _grpc_certificate_path)
+                    stub_method = grpc_helper_obj.get_method_from_stub_object(stub_class_obj, _grpc_communication_type,
+                                                                              stub_name=_grpc_rpc_stub_method_name)
+                    if grpc_function_json_data:
```

Comment:
When will `grpc_function_json_data` be `None`? Is there a need for this check?

Reply:
It can be None, if user doesn't pass or give the proper pre-process script.

Reply:
If the user passes the incorrect JSON String format, line 126: `grpc_function_json_data = json.loads(grpc_function_data)` will fail and an Exception will be raised. So I don't think this `if` statement is needed